application/usecase/pokemon.go

```diff
@@ -0,0 +1,82 @@
+package usecase
+
+import (
+	"fmt"
+
+	"github.com/FernandoGal25/academy-go-q42021/application/contracts"
+	"github.com/FernandoGal25/academy-go-q42021/domain/model"
+	errors "github.com/FernandoGal25/academy-go-q42021/error"
+)
+
+const totalPokemon = 898
+
+// Service that handles the pokemon case uses.
+type PokemonService struct {
+	CSVPokemonRepository  contracts.PokemonRepository
+	RestPokemonRepository contracts.PokemonRepository
+}
+
+// Returns an instance of PokemonService.
+func NewPokemonService(r1 contracts.PokemonRepository, r2 contracts.PokemonRepository) PokemonService {
+	return PokemonService{r1, r2}
+}
+
+func validateRangeID(key int) error {
+	if key > totalPokemon {
+		return errors.ErrDomainValidation{
+			Message: fmt.Sprintf("Out of bound, pokemon with ID: %v searched while there are currently only %v pokemon.", key, totalPokemon),
```

Comment:
Even this is not using an already generated error you could initialize one and make use here. It's always a good idea to be consistent in returning an error object so it can be wrapped by the previous layer(s).

Could be an error defined in error package, or at the same usecase level.
```suggestion
			Message: fmt.Sprintf("Out of bound, pokemon with ID: %v searched while there are currently only %v pokemon.", key, totalPokemon),
			Err: pokemonError.ErrOutOfBounds,
```